harness/determined/common/experimental/trial.py

```diff
@@ -125,6 +125,13 @@ class Trial:
         trial_id: ID of trial.
         session: HTTP request session.
         hparams: (Mutable, Optional[Dict]) Hyperparameters for the trial.
+        summary: (Mutable, Optional[Dict]) Summary metrics for the trial.
```

Comment:
I find it frustrating that we don't have documentation to refer to for what hyperparameter objects or summary metrics objects are. How is a user supposed to know what to do with these?

Reply:
It's clearly not your job with this PR, but I want it to be _someone's_ job, and I care enough to drive it. Before I dive into getting docs about summary metrics published, do you happen to know if there's something already there I'm just missing?

Reply:
i added more detailed docstrings and will follow up on whether this warrants a more full-fledged user-facing documentation